python/polars_ds/num.py

```diff
@@ -1066,3 +1066,43 @@ def _haversine(
             is_elementwise=True,
             cast_to_supertypes=True,
         )
+
+    def knn_entropy(self, *others: pl.Expr, k=2):
+        """
+        Multivariate estimation of entropy through a k-nearest neighbor query
+
+        Reference
+        ---------
+        https://arxiv.org/pdf/1506.06501v1.pdf
+        """
+        from scipy.special import digamma
+        from math import pi, gamma, log
+        (N, d) = pl.len(), len(others) + 1  # d has to include self
+        g1 = digamma(N) - digamma(k)
+        knn = pl.col('row_nr').num.knn_ptwise(*others, dist='euclidean', k=k).list.get(k)
```

Comment:
Right now I think dist = 'l2' is available, but l2 is actually squared Euclidean. But that won't change the output because sqrt preserves order..

Also I think pl.col("row_nr") should be self._expr?

Reply:
ow you are 100% right